<?xml version="1.0"?>
<launch>
	<arg name="aruco_size"					default="0.08" />
	<arg name="windows_reduction"			default="2" />
	<arg name="camera_name"					default="bebop_camera" />
	
	<!-- joy node -->
	<node respawn="true" pkg="joy"
		type="joy_node" name="turtle_joy" >
	<param name="dev" type="string" value="/dev/input/js0" />
	<param name="deadzone" value="0.12" />
	</node>

	<!-- Axes -->
	<node pkg="light_inspector" type="turtle_teleop_joy" name="teleop" />
	
	<!-- Joy_Bebop_Interface -->
	<node pkg="light_inspector" type="joy_bebop_interface" name="interface"/>
	
	<!-- Image View Node -->
<!--
	<node pkg="image_view" type="image_view" name="bebop_image">
		<remap from="image" to="/bebop/image_raw"/>
		<param name="autosize" value="true" type="bool"/>
	</node>
-->
	
	<!-- Image View Node -->
<!--
	<node pkg="light_inspector" type="image_view" name="aruco_pose_estimation">
	</node>
-->
	
	<!-- Aruco Pose Estimation Node -->
	<node pkg="light_inspector" type="aruco_pose_estimation" name="bebop_image" output="screen">
		<param name="aruco_size"   			type="double"   value="$(arg aruco_size)"/> 
		<param name="windows_reduction"   	type="int"   	value="$(arg windows_reduction)"/> 
		<param name="camera_name"   		type="string"   value="$(arg camera_name)"/> 
	</node>
	
	<!-- Video Record Node -->
<!--
	<node pkg="image_view" type="video_recorder" name="bebop_record" output="screen">
		<remap from="image" to="/bebop/image_raw"/>
		<param name="fps" value="30" type="int"/>
	</node>
-->
	
	<!-- Bebeop Driver Node -->
	<arg name="namespace" default="bebop" />
    <arg name="ip" default="192.168.42.1" />
    <arg name="drone_type" default="bebop2" /> <!-- available drone types: bebop1, bebop2 -->
    <arg name="config_file" default="$(find bebop_driver)/config/defaults.yaml" />
    <arg name="camera_info_url" default="package://bebop_driver/data/$(arg drone_type)_camera_calib.yaml" />
    <group ns="$(arg namespace)">
        <node pkg="bebop_driver" name="bebop_driver" type="bebop_driver_node">
            <param name="camera_info_url" value="$(arg camera_info_url)" />
            <param name="bebop_ip" value="$(arg ip)" />
            <rosparam command="load" file="$(arg config_file)" />
        </node>
        <include file="$(find bebop_description)/launch/description.launch" />
    </group>
    
</launch>
